src/main/java/SoloTech/StockFlow/common/aop/CacheAspect.java

```diff
@@ -0,0 +1,131 @@
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.reflect.MethodSignature;
+import org.springframework.core.DefaultParameterNameDiscoverer;
+import org.springframework.core.ParameterNameDiscoverer;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.expression.spel.support.StandardEvaluationContext;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.Method;
+import java.time.Duration;
+
+/**
+ * 캐시관점 : 코드 중복 줄이고, 유지보수성 높이며, 공통 관심사를 관리
+ */
+
+@Slf4j
+@Aspect
+@Component
+@RequiredArgsConstructor
+public class CacheAspect {
+    private final Cache<String,Object> localCache;
+    private final RedisTemplate<String,Object> redisTemplate;
+    private final CachePublisher cachePublisher;
+    private final SpelExpressionParser parser = new SpelExpressionParser();
+    private static final long DEFAULT_TTL = 60;
+
+    @Around("@annotation(cached)")
+    public Object handleCaching(ProceedingJoinPoint joinPoint, Cached cached) throws Throwable{
+        String prefix = cached.prefix();
+        String keyExpression = cached.key();
+        long ttl = cached.ttl();
+        CacheType type = cached.type();
+
+        // TTL 검증 및 기본값 설정
+        if (ttl <= 0) {
+            log.warn("Invalid TTL value: {}. Using default TTL.", ttl);
+            ttl = DEFAULT_TTL;
+        }
+
+        MethodSignature signature = (MethodSignature) joinPoint.getSignature();
+        Method method = signature.getMethod();
+        String key = buildKeyFromSpEL(keyExpression, method, joinPoint.getArgs());
+
+        String cacheKey = prefix + key;
+
+        Object result;
+
+        switch (type) {
+            case READ -> {
+                Object localValue = localCache.getIfPresent(cacheKey);
+                if (localValue != null) return localValue;
+
+                Object redisValue = redisTemplate.opsForValue().get(cacheKey);
+                if (redisValue != null) {
+                    localCache.put(cacheKey, redisValue);
+                    return redisValue;
+                }
+
+                result = joinPoint.proceed();
+                if (result != null) {
+                    redisTemplate.opsForValue().set(cacheKey, result, Duration.ofSeconds(ttl));
+                    localCache.put(cacheKey, result);
+                    log.info("[READ] Cached: {}", cacheKey);
+                }
+                return result;
+            }
+
+            case WRITE -> {
+                result = joinPoint.proceed();
+                if (result != null) {
+                    redisTemplate.opsForValue().set(cacheKey, result, Duration.ofSeconds(ttl));
+                    localCache.put(cacheKey, result);
+
+                    String message = "Updated " + prefix+ "-" + cacheKey;
```

Comment:
```suggestion
                    String message = "UPDATE " + prefix+ "-" + cacheKey;
```
update, delete 포맷이 다른데요

각각의 요소를 enum 으로 관리 하고

캐시키를 생성하는  util 을 만드는것도 좋을것 같습니다

Reply:
공용 util로 따로 빼두록 하겠습니다!